File: Data-Structures/Stack/StackUsingArray.js
class Stack {
  constructor() {
    this.stack = [];
    this.top = -1;
  }
  push(value) {
    this.top++;
    this.stack[this.top] = value;
  }
  pop() {
    if (this.top == -1) {
      return "Stack is empty";
    }
    var ret_val = this.stack[this.top];
    this.top--;
    return ret_val;
  }

  peek() {
    if (this.top == -1) {
      return "Stack is empty";
    }
    var ret_val = this.stack[this.top];
    return ret_val;
  }

  isEmpty() {
    return this.top == -1;
  }
}

stk = new Stack();
console.log(stk.isEmpty());
stk.push(1);
stk.push(2);
console.log(stk.peek());
console.log(stk.isEmpty());
stk.push(3);
stk.push(4);
console.log(stk.pop());
console.log(stk.pop());
console.log(stk.pop());
console.log(stk.pop());
console.log(stk.isEmpty());

File: Data-Structures/Stack/StackUsingArray.cs
using System;
using System.Collections.Generic;

namespace Stack
{
  class Program
  {
    static void Main(string[] args)
    {
      Stack stack = new Stack();
      Console.WriteLine(stack.isEmpty());
      for (int i = 1; i < 6; i++)
      {
        stack.Push(i);
      }
      Console.WriteLine(stack.isEmpty());
      stack.Peek();
      for (int i = 1; i < 6; i++)
      {
        stack.Pop();
      }
      Console.WriteLine(stack.isEmpty());
    }
  }

  class Stack
  {
    List<int> stack = new List<int>();
    int top;

    public Stack()
    {
      top = -1;
    }

    public void Push(int value)
    {
      stack.Add(value);
      top++;
    }

    public void Pop()
    {
      if (isEmpty())
      {
        Console.WriteLine("Stack is empty!");
      }
      int value = stack[top];
      stack.RemoveAt(top);
      top--;
      Console.WriteLine($"The value {value} was removed of stack");
    }

    public void Peek()
    {
      if (isEmpty())
      {
        Console.WriteLine("Stack is empty!");
      }
      Console.WriteLine($"The topmost element of stack is {stack[top]}");
    }

    public bool isEmpty()
    {
      return top == -1;
    }
  }
}
